### PR TITLE
Improve styling and wording of SSO redirect confirm template

### PR DESCRIPTION
**Pull request by richvdh**
~~Based on #9271.~~

### CODE REVIEW

File: synapse/module_api/__init__.py [hunk 1]
```diff
@@ -279,7 +279,11 @@ def complete_sso_login(
         )
 
     async def complete_sso_login_async(
-        self, registered_user_id: str, request: SynapseRequest, client_redirect_url: str
+        self,
+        registered_user_id: str,
+        request: SynapseRequest,
+        client_redirect_url: str,
+        new_user: bool = False,
```
Comment (clokep) [inline synapse/module_api/__init__.py:286]: Does it make sense to add this to the `complete_sso_login` version as well? I'm not sure what even calls these anymore.

File: synapse/res/templates/sso_redirect_confirm.html [hunk 2]
```diff
@@ -3,12 +3,34 @@
...
+    </style>
 </head>
     <body>
-        <p>The application at <span style="font-weight:bold">{{ display_url | e }}</span> is requesting full access to your <span style="font-weight:bold">{{ server_name }}</span> Matrix account.</p>
-        <p>If you don't recognise this address, you should ignore this and close this tab.</p>
-        <p>
-            <a href="{{ redirect_url | e }}">I trust this address</a>
-        </p>
+        <header>
+            <!-- {% if new_user %} -->
```
Comment (clokep) [inline synapse/res/templates/sso_redirect_confirm.html:13]: Why putting this in HTML comment tags?

File: synapse/res/templates/sso_redirect_confirm.html [hunk 3]
```diff
@@ -3,12 +3,34 @@
...
+            <h1>Your account is now ready</h1>
+            <p>You've made your account on {{ server_name | e }}.</p>
```
Comment (clokep) [inline synapse/res/templates/sso_redirect_confirm.html:15]: With autoescape on I don't think we should be manually escaping these as it is confusing (although it shouldn't cause an error). (Or is #9200 not on the social login branch?)

File: synapse/res/templates/sso_redirect_confirm.html [hunk 4]
```diff
@@ -3,12 +3,34 @@
...
+            <!-- {% else %} -->
+            <h1>Log in</h1>
+            <!-- {% endif %} -->
+            <p>Continue to confirm you trust <strong>{{ display_url | e }}</strong>.</p>
+        </header>
+        <main>
+            <!-- {% if user_profile.avatar_url and user_profile.display_name %} -->
```
Comment (clokep) [inline synapse/res/templates/sso_redirect_confirm.html:22]: Should this code account for having only a display name and not an avatar?

File: synapse/module_api/__init__.py (hunk 1 above)
Comment (richvdh) [reply synapse/module_api/__init__.py:286]: given it's deprecated, I'm not sure there's any value in updating it. Any modules which want to modify the value of `new_user` should call the async version.

File: synapse/res/templates/sso_redirect_confirm.html (hunk 2 above)
Comment (richvdh) [reply synapse/res/templates/sso_redirect_confirm.html:13]: to make it it easier to develop outside the template engine, I believe.

File: synapse/res/templates/sso_redirect_confirm.html (hunk 3 above)
Comment (richvdh) [reply synapse/res/templates/sso_redirect_confirm.html:15]: > (Or is #9200 not on the social login branch?)

yup, that, I'm afraid.

File: synapse/res/templates/sso_redirect_confirm.html (hunk 4 above)
Comment (richvdh) [reply synapse/res/templates/sso_redirect_confirm.html:22]: the design is not to show this section at all if we don't have an avatar (which we never will, currently).

That said, it ought to account for having an avatar and not a display name...

File: synapse/res/templates/sso_redirect_confirm.html (hunk 2 above)
Comment (clokep) [reply synapse/res/templates/sso_redirect_confirm.html:13]: I've never seen this pattern before. What makes it easier to develop without these? We don't seem to do this in any other templates. Personally I find it makes my eyes ignore the Jinja logic.

**Pull request merged**